crates/pallet-domains/src/tests.rs

```diff
@@ -820,6 +823,73 @@ fn storage_proof_for_key<T: Config, B: Backend<T::Hashing> + AsTrieBackend<T::Ha
     (root, proof)
 }
 
+#[test]
+fn test_invalid_domain_extrinsic_root_proof() {
+    let creator = 0u64;
+    let operator_id = 1u64;
+    let head_domain_number = 10;
+    let mut ext = new_test_ext_with_extensions();
+    ext.execute_with(|| {
+        let domain_id = register_genesis_domain(creator, vec![operator_id]);
+        extend_block_tree(domain_id, operator_id, head_domain_number + 1);
+        assert_eq!(
+            HeadReceiptNumber::<Test>::get(domain_id),
+            head_domain_number
+        );
+
+        let bad_receipt_at = 8;
+        let domain_block = get_block_tree_node_at::<Test>(domain_id, bad_receipt_at).unwrap();
+
+        let bad_receipt_hash = domain_block.execution_receipt.hash();
+        let (fraud_proof, root) = generate_invalid_domain_extrinsic_root_fraud_proof::<Test>(
+            domain_id,
+            bad_receipt_hash,
+            Randomness::from([1u8; 32]),
+        );
```

Comment:
This is not legitimate, it basically creates valid fraud proof that targets a valid ER, I fixed this in https://github.com/subspace/subspace/pull/1983/commits/44d3a69c244c7356e6e0ee7b7d9b0e28e86bd337 though.